subworkflows/ebi-metagenomics/reads_bwamem2_decontamination/main.nf

```diff
@@ -0,0 +1,26 @@
+include { BWAMEM2_MEM     } from '../../../modules/ebi-metagenomics/bwamem2/mem/main'
+include { SAMTOOLS_BAM2FQ } from '../../../modules/ebi-metagenomics/samtools/bam2fq/main'
+
+
+workflow READS_BWAMEM2_DECONTAMINATION {
+
+    take:
+    ch_reads       // channel (mandatory): [ val(meta), [ path(reads) ] ]
+    ch_reference   // channel (mandatory): [ val(meta2), path(ref_index) ] | meta2 contains the name of the reference genome
+
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    BWAMEM2_MEM(ch_reads, ch_reference)
+    ch_versions = ch_versions.mix(BWAMEM2_MEM.out.versions.first())
+
+    SAMTOOLS_BAM2FQ( BWAMEM2_MEM.out.bam.map { meta, bam, _ -> [ meta, bam, meta.single_end == false ] } )
```

Comment:
Have we decided to remove all the samtools view filters?

Reply:
No, they will be passed through configs

[either in nextflow.config, or better, in the module config:
```
process {
    ext.args = "-f 12 -F 256 -uS"
}
```
]